### ISSUE TITLE
unknown parameters in lnd.conf throw an unrelated error

### ISSUE THREAD

**Issue by Deelight-fr** [lightningnetwork/lnd#652]
Specifying an unknown parameter in lnd.conf generally throws an unrelated error:

`loadConfig: either bitcoin.active or litecoin.active must be set to 1 (true)`

The problem is inconsistent. Restarting lnd a few time may result in a diffent (an correct) error : `unknown option`

Example (`foo=bar` added to lnd.conf):

```
$ lnd
loadConfig: either bitcoin.active or litecoin.active must be set to 1 (true)
$ lnd
loadConfig: either bitcoin.active or litecoin.active must be set to 1 (true)
$ lnd
loadConfig: either bitcoin.active or litecoin.active must be set to 1 (true)
$ lnd
loadConfig: either bitcoin.active or litecoin.active must be set to 1 (true)
$ lnd
loadConfig: either bitcoin.active or litecoin.active must be set to 1 (true)
$ lnd
loadConfig: either bitcoin.active or litecoin.active must be set to 1 (true)
$ lnd
loadConfig: either bitcoin.active or litecoin.active must be set to 1 (true)
$ lnd
loadConfig: either bitcoin.active or litecoin.active must be set to 1 (true)
$ lnd
2018-01-22 10:46:30.905 [WRN] LTND: /home/deelight/.lnd/lnd.conf:9: unknown option: foo
2018-01-22 10:46:30.905 [INF] LTND: Version 0.3.0-alpha
2018-01-22 10:46:30.906 [INF] CHDB: Checking for schema update: latest_version=0, db_version=0
2018-01-22 10:46:31.035 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
2018-01-22 10:46:31.035 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
2018-01-22 10:46:31.035 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create wallet, or `lncli unlock` to unlock already created wallet.
```

**Comment by MeshCollider** [comment]
I believe this is because the ini parser returns if it encounters an unknown option, rather than continuing to pass the rest first:
https://github.com/jessevdk/go-flags/blob/f88afde2fa19a30cf50ba4b05b3d13bc6bae3079/ini.go#L531

Its random because the INI reader uses a map so the ordering of parameters being processed is not deterministic I assume.

Then, the error message is returned but is not output until the very end of loadConfig(), `// Warn about missing config file only after all other configuration is done.  This prevents the warning on help messages and invalid options.  Note this should go directly before the return.`
Unfortunately this means the code will terminate early a lot of the time without ever printing out the warning as you said, because the other parameters from the file haven't been loaded and appear missing to lnd

**Comment by lucash-dev** [comment]
"This prevents the warning on help messages and invalid options." -- This behaviour seems to be wrong. 
Wouldn't it make more sense to have parse errors in the ini file to stop execution? Help messages are another case though.

**Comment by PWKad** [comment]
Similarly if you don't set `litecoind.rpcuser` the error messages are hard-coded in `config.go` to show `bitcoind.rpcuser` even though the configuration is "shared".  I'll try to put together a PR to fix that as well.

**Comment by PWKad** [comment]
Fixed by #916
